backend/lucene/src/main/java/org/hibernate/search/backend/lucene/search/predicate/impl/AbstractLuceneSearchPredicateBuilder.java

```diff
@@ -52,4 +54,22 @@ public final Query build(LuceneSearchPredicateContext context) {
 	}
 
 	protected abstract Query doBuild(LuceneSearchPredicateContext context);
+
+	protected Query applyImplicitNestedSteps(List<String> furtherImplicitNestedSteps, LuceneSearchPredicateContext context,
+			Function<LuceneSearchPredicateContext, Query> baseBuild) {
+		if ( furtherImplicitNestedSteps.isEmpty() ) {
+			return baseBuild.apply( context );
+		}
+
+		if ( furtherImplicitNestedSteps.size() == 1 ) {
+			String lastStep = furtherImplicitNestedSteps.get( 0 );
+			LuceneSearchPredicateContext childContext = new LuceneSearchPredicateContext( lastStep );
+			// super.build( childContext ) must be called only from the very last nest step
+			return LuceneNestedPredicateBuilder.doBuild( context, lastStep, baseBuild.apply( context ) );
```

Comment:
You don't use the child context. This seems wrong, and there's probably a missing test, otherwise this would have been detected by a failing tests.

Reply:
Oh yes, I don't. Thank you. I'm going to add a test for that.

Reply:
Ok, I checked it...

Basically, the passed (child) context would be used by another implicit or explicit nested predicates.
So if there were a subsequent nest, it would benefit from it.
I think it's not impossible to have any nested after (nested) an implicit nested, because implicit nested always reaches the leaf (field) of the tree.
But let me know if you agree with that.

Anyway, I have to remove the unused local variable, so I'll do it.

Reply:
That is the same reason why `AbstractElasticsearchSearchNestedPredicateBuilder#applyImplicitNestedStep` does not need to update the context.

Reply:
Agreed, it makes sense.